src/org/zaproxy/zap/extension/pscan/scanner/RegexAutoTagScanner.java

```diff
@@ -310,9 +310,17 @@ public boolean equals(Object object) {
         return true;
     }
     
-	private void matched(HttpMessage msg, int id) {
+	private void matched(Matcher matcher, HttpMessage msg, int id) {
+		String tag = getConf();
 		if (tagHistoryType(msg.getHistoryRef().getHistoryType())) {
-			parent.addTag(id, this.getConf());
+			if (matcher.groupCount() > 0) {
+				StringBuffer result = new StringBuffer();
+				matcher.appendReplacement(result, tag);
+				tag = result.substring(matcher.start());
+				parent.addTag(id, tag);
```

Comment:
This could be done outside/after the if/then branches.

Should the replaced tag also be used in the stats?

Reply:
Good question. I would think based on the tag firing not necessarily based on each variation if the tag. But that's just a quick gutt reaction.

Reply:
Yeah, makes sense.